packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/licence-summary.spec.js

```diff
@@ -58,249 +38,97 @@ afterAll(() => {
-    const data = await injectWithCookies('GET', DATE_OF_BIRTH.uri)
-    expect(data.payload.search(backLinkRegEx(LICENCE_SUMMARY.uri)) > 0).toBeTruthy()
-    const data2 = await postRedirectGet(DATE_OF_BIRTH.uri, dobHelper(dobAdultToday))
-    expect(data2.statusCode).toBe(302)
-    expect(data2.headers.location).toBe(LICENCE_SUMMARY.uri)
-  })
-
-  it('date of birth (senior) amendment causes redirect the licence summary page', async () => {
-    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobSeniorToday))
-    const data = await injectWithCookies('GET', CONTROLLER.uri)
-    expect(data.statusCode).toBe(302)
-    expect(data.headers.location).toBe(LICENCE_SUMMARY.uri)
-  })
-
-  it('date of birth amendment (no licence required) causes a redirect to the no licence required page', async () => {
-    await postRedirectGet(LICENCE_TO_START.uri, { 'licence-to-start': 'after-payment' })
-    const data = await postRedirectGet(DATE_OF_BIRTH.uri, dobHelper(dobJuniorTomorrow))
-    expect(data.statusCode).toBe(302)
-    expect(data.headers.location).toBe(NO_LICENCE_REQUIRED.uri)
-  })
-
-  it('date of birth amendment (junior) causes redirect to the junior licence page and subsequent redirect to the licence summary page and then the contact summary page', async () => {
-    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobJuniorToday))
-    let data = await injectWithCookies('GET', CONTROLLER.uri)
-    expect(data.statusCode).toBe(302)
-    expect(data.headers.location).toBe(JUNIOR_LICENCE.uri)
-    data = await postRedirectGet(JUNIOR_LICENCE.uri, {})
-    expect(data.statusCode).toBe(302)
-    expect(data.headers.location).toBe(LICENCE_SUMMARY.uri)
-  })
-
-  it('Posting the licence summary page redirects to the name page', async () => {
-    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
-    const data = await postRedirectGet(LICENCE_SUMMARY.uri)
-    expect(data.statusCode).toBe(302)
-    expect(data.headers.location).toBe(NAME.uri)
-  })
-
-  it('date of birth amendment (junior) causes a method of contact of letter to be set to none', async () => {
-    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobSeniorToday))
-    await injectWithCookies('GET', CONTROLLER.uri)
-    await postRedirectGet(CONTACT.uri, { 'how-contacted': 'none' })
-    await injectWithCookies('POST', DATE_OF_BIRTH.uri, dobHelper(dobJuniorToday))
-    await injectWithCookies('GET', CONTROLLER.uri)
-    const { payload } = await injectWithCookies('GET', TEST_TRANSACTION.uri)
-    expect(JSON.parse(payload).permissions[0].licensee.preferredMethodOfConfirmation).toEqual(HOW_CONTACTED.none)
-    expect(JSON.parse(payload).permissions[0].licensee.preferredMethodOfReminder).toEqual(HOW_CONTACTED.none)
-    expect(JSON.parse(payload).permissions[0].licenceLength).toBe('12M')
+  describe('where the prerequisite are not fulfilled', async () => {
+    beforeAll(async d => {
+      await injectWithCookies('GET', CONTROLLER.uri)
+      d()
```

Comment:
Is this necessary in an async method? Thought it was only useful in a sync method where you're passing callbacks

Reply:
I believe so. The method is async because it contains awaits but is not actually async internally meaning that a callback is required. I experimented with it some while ago but I believe that to be the case